core/src/main/resources/lib/form/filter-menu-button/filter-menu-button.js

```diff
@@ -0,0 +1,56 @@
+function createFilterMenuButton (button, menu, menuAlignment, menuMinScrollHeight) {
+    var MIN_NUM_OPTIONS = 5;
+    var menuButton = new YAHOO.widget.Button(button, {
+        type: "menu",
+        menu: menu,
+        menualignment: menuAlignment,
+        menuminscrollheight: menuMinScrollHeight
+    });
+
+    var filter = _createFilter(menuButton._menu);
+
+    menuButton._menu.element.appendChild(filter);
+    menuButton._menu.showEvent.subscribe(function() {
+        filter.firstElementChild.value = '';
+        _applyFilterKeyword(menuButton._menu, filter.firstElementChild);
+        filter.style.display = (_getItemList(menuButton._menu).children.length >= MIN_NUM_OPTIONS) ? '' : 'NONE';
+    });
+    menuButton._menu.setInitialFocus = function () {
+        setTimeout(function() {
+            filter.firstElementChild.focus();
+        }, 0);
+    };
+
+    return menuButton;
+}
+
+function _createFilter (menu) {
+    var filterInput = document.createElement("input");
+    filterInput.style.width = '100%';
+    filterInput.setAttribute("placeholder", "Filter");
```

Comment:
```suggestion
    filterInput.setAttribute("placeholder", "Filter");
    filterInput.setAttribute("spellcheck", "false");
    filterInput.setAttribute("type", "search");
```

This fixed the spellcheck problem for me, also uses the correct input type so you get a nice little ⓧ button for clearing text 👍